catalog/model/src/main/java/org/projectnessie/catalog/model/ops/CatalogOperation.java

```diff
@@ -31,4 +31,12 @@ public interface CatalogOperation {
   ContentKey getKey();
 
   Content.Type getType();
+
+  /**
+   * The logical warehouse name where this operation will occur. Must correspond to a warehouse
+   * configured under {@code nessie.catalog.warehouses.<name>}.
+   *
+   * <p>If not set, the default warehouse will be used.
+   */
+  String warehouse();
```

Comment:
That was my first impl, but in fact there is no need of that because of how `CatalogConfig.getWarehouse` is implemented: it already handles nulls gracefully. I should have added `@Nullable` here though, I will fix that.

Reply:
WFM